src/RealtimeClient.ts

```diff
@@ -360,28 +361,14 @@ export default class RealtimeClient {
       (this.accessToken && (await this.accessToken())) ||
       this.accessTokenValue
 
-    if (tokenToSend) {
-      let parsed = null
-      try {
-        parsed = JSON.parse(atob(tokenToSend.split('.')[1]))
-      } catch (_error) {}
-      if (parsed && parsed.exp) {
-        let now = Math.floor(Date.now() / 1000)
-        let valid = now - parsed.exp < 0
-        if (!valid) {
-          this.log(
-            'auth',
-            `InvalidJWTToken: Invalid value for JWT claim "exp" with value ${parsed.exp}`
-          )
-          return Promise.reject(
-            `InvalidJWTToken: Invalid value for JWT claim "exp" with value ${parsed.exp}`
-          )
-        }
-      }
-
+    if (this.accessTokenValue != tokenToSend) {
       this.accessTokenValue = tokenToSend
       this.channels.forEach((channel) => {
-        tokenToSend && channel.updateJoinPayload({ access_token: tokenToSend })
+        tokenToSend &&
+          channel.updateJoinPayload({
+            access_token: tokenToSend,
+            version: VERSION,
```

Comment:
@filipecabaco instead of sending version, would it be better to send the x-client-info? To differentiate which client is being used?